src/test/java/io/vertx/test/fakecluster/FakeClusterManager.java

```diff
@@ -207,107 +203,162 @@ public void leave(Handler<AsyncResult<Void>> resultHandler) {
         }
       }
       fut.complete();
-    }, resultHandler);
+    });
   }
 
   @Override
   public boolean isActive() {
     return nodeID != null;
   }
 
+  @Override
+  public Future<Void> register(String address, RegistrationInfo registrationInfo) {
+    registrations.computeIfAbsent(address, k -> Collections.synchronizedList(new ArrayList<>()))
```

Comment:
instead use a non plain unmodifiable list in the registrations map that is modified using compute. Otherwise there can be races with concurrent unregistration, see the comment below.